torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -61,6 +61,12 @@ std::ostream& operator<<(std::ostream & out, const TensorDesc & d) {
 namespace codegen {
 
 auto type_declarations_template = CodeTemplate(R"(
+#if defined(__CUDACC_RTC__)
+typedef unsigned char uint8_t; 
+typedef signed char int8_t; 
+typedef short int  int16_t; 
+typedef long long int int64_t; 
```

Comment:
Actually why do we even need those typedefs?

Reply:
Is it really impossible to include `stdint.h` from NVRTC? Seems like quite a basic and important header. If that's the case then let's leave it, but it would be better to avoid those hacks.

Reply:
With nvrtc, you have to be sure stdint.h will be there on the users system when nvrtc is compiling stuff, and you know where to find it on the users system, which in practice means pytorch has to redistribute stdint.h. I don't think anyone wants it.   
I agree it's better to avoid those hacks, and that's why we are working on getting cuda fp16 headers to be redistributable, so that when jitter supports half, if won't also have to have type defs and intrinsics in the form of copypasta, and can rely on fp16 headers being redistributed with pytorch. However, with integer types the problem seems manageable enough with 4 lines of typedefs.

Reply:
Makes sense, thanks for the explanation!

Reply:
Would be nice to have this conversation as a comment in the code ;)